File: libSOR/src/general/PedidoCorto.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */

package general;

/**
 *
 * @author Cute
 */
public class PedidoCorto {
     private String ID;
     private EstadoPedido estado;

    public PedidoCorto(String ID, EstadoPedido estado) {
        this.ID = ID;
        this.estado = estado;
    }
    public PedidoCorto(String ID,String estado){
        this.ID=ID;
        this.estado= EstadoPedido.values()[Integer.parseInt(estado)];
    }
    public PedidoCorto(){
        this.ID="";
        this.estado=null;
    }

    public PedidoCorto(PedidoCorto p){
    this.ID=p.getID();
    this.estado=p.getEstado();
    }
    public String getID() {
        return ID;
    }

    public void setID(String ID) {
        this.ID = ID;
    }

    public EstadoPedido getEstado() {
        return estado;
    }

    public void setEstado(EstadoPedido estado) {
        this.estado = estado;
    }
     
}

File: desguaceNET/desguaceNET/libSOR/general/PedidoCorto.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace desguaceNET.libSOR.general
{
    class PedidoCorto
    {
        public string ID {get;set;}
        public EstadoPedido estado { get; set; }

        public PedidoCorto()
        {
            this.ID = "";
            this.estado = EstadoPedido.NEW;
        }
        public PedidoCorto(string ID, EstadoPedido estado)
        {
            this.ID = ID;
            this.estado = estado;
        }
        public PedidoCorto(string ID, string estado)
        {
            this.ID = ID;
            this.estado = (EstadoPedido) Enum.Parse(typeof(EstadoPedido),estado);
        }

        public PedidoCorto(PedidoCorto p)
        {
            this.ID = p.getID();
            this.estado = p.getEstado();
        }
        public string getID()
        {
            return ID;
        }

        public void setID(string ID)
        {
            this.ID = ID;
        }

        public EstadoPedido getEstado()
        {
            return estado;
        }

        public void setEstado(EstadoPedido estado)
        {
            this.estado = estado;
        }
    }
}
